clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -500,15 +500,24 @@ public StoreIngestionTask(
         storageService.getRefCountedStorageEngine(storeVersionName);
     this.thingsToClose.add(refCountedStorageEngine);
     this.storageEngine = Objects.requireNonNull(refCountedStorageEngine.get());
+    this.isDaVinciClient = builder.isDaVinciClient();
+
     if ((this.storageEngine instanceof DelegatingStorageEngine)) {
       DelegatingStorageEngine delegatingStorageEngine = (DelegatingStorageEngine) this.storageEngine;
-      delegatingStorageEngine.setKeyDictCompressionFunction(p -> {
-        PartitionConsumptionState pcs = partitionConsumptionStateMap.get(p);
-        if (pcs == null) {
-          throw new VeniceException("Partition " + p + " not found in partitionConsumptionStateMap");
-        }
-        return pcs.getKeyDictCompressor();
-      });
+
+      if (isDaVinciClient) {
+        delegatingStorageEngine.setKeyDictCompressionFunction(p -> {
+          PartitionConsumptionState pcs = partitionConsumptionStateMap.get(p);
+          if (pcs == null) {
+            throw new VeniceException("Partition " + p + " not found in partitionConsumptionStateMap");
+          }
+          return pcs.getKeyDictCompressor();
+        });
+      } else {
+        // Key Compression is only enabled in Venice Server.
```

Comment:
Shouldn't be `Key Compression is only available in DVC`?

Reply:
Yeah that's typo. Fixed.